modules/kubernetes/metrics/jobs/agent.river

```diff
@@ -0,0 +1,238 @@
+/*
+Module: job-agent
+Description: Scrapes grafana agent
+
+Note: Every argument except for "forward_to" is optional, and does have a defined default value.  However, the values for these
+      arguments are not defined using the default = " ... " argument syntax, but rather using the coalesce(argument.value, " ... ").
+      This is because if the argument passed in from another consuming module is set to null, the default = " ... " syntax will
+      does not override the value passed in, where coalesce() will return the first non-null value.
+*/
+argument "forward_to" {
+  comment = "Must be a list(MetricsReceiver) where collected logs should be forwarded to"
+}
+
+argument "enabled" {
+  comment = "Whether or not the grafana-agent job should be enabled, this is useful for disabling the job when it is being consumed by other modules in a multi-tenancy environment"
+  optional = true
+}
+
+argument "namespaces" {
+  comment = "The namespaces to look for targets in (default: [] is all namespaces)"
+  optional = true
+}
+
+argument "selectors" {
+  // Docs: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
+  comment = "The label selectors to use to find matching targets (default: [\"app.kubernetes.io/name=grafana-agent\"])"
+  optional = true
+}
+
+argument "port_name" {
+  comment = "The of the port to scrape metrics from (default: http-metrics)"
+  optional = true
+}
+
+argument "job_label" {
+  comment = "The job label to add for all grafana-agent metric (default: integrations/agent)"
+  optional = true
+}
+
+argument "keep_metrics" {
+  comment = "A regex of metrics to keep (default: see below)"
+  optional = true
+}
+
+argument "scrape_interval" {
+  comment = "How often to scrape metrics from the targets (default: 60s)"
+  optional = true
+}
+
+argument "scrape_timeout" {
+  comment = "How long before a scrape times out (default: 10s)"
+  optional = true
+}
+
+argument "max_cache_size" {
+  comment = "The maximum number of elements to hold in the relabeling cache (default: 100000).  Only increase if the amount of metrics returned is extremely large, the default will almost always be sufficient"
```

Comment:
Change comment that it should be at least 2x-5x your largest scrape target or samples appended rate. Been playing with this some